tests_e2e/tests/lib/virtual_machine.py

```diff
@@ -1,155 +0,0 @@
-# Microsoft Azure Linux Agent
```

Comment:
The initial prototype tried to define a common interface for Virtual Machine and Scale Set operations, and I tried to maintain that model in my first implementation of this code.

However, that model does not quite work. Though there are some similarities in the APIs and the invocation patterns, the differences and significant enough to make a common implementation awkward. I already had several TODO comments to revisit that decision. Since I needed to change those classes to support the update operation, I decided to revisit the design now. I removed the implementation for scale sets and merged the implementation of the base class into VirtualMachine.

I renamed VirtualMachine to VirtualMachineClient, because the former conflicts with class VirtualMachine in the SDK (the get_description method, needed by #2810, returns an SDK's VirtualMachine).

I made similar changes on VmExtension (now VirtualMachineExtensionClient).

@nagworld9 This refactoring affects #2810, but the impact in your test should be minimal (just a few renames).

Reply:
leaving a question here that may help me with migrating my current dcr scenario (ext-workflow)

in the dcr repo we have classes to define each extension we use for testing (https://msazure.visualstudio.com/One/_git/Compute-Runtime-Tux-Pipeline?path=/dungeon_crawler/scenarios/extensions.py)

We're currently using the GuestAgentTestDcr extension for testing, and I'm trying to decide the best place to define a similar class in the new e2e pipeline. 

I haven't seen any similar class made for RunCommand or CSE yet in the e2e pipeline, do you have any recommendations on where/if I should make this class. I'm wondering if it should be an implementation of virtual machine extension client? Let me know if you have any ideas

Reply:
The new code base is organized differently: extension operations are defined in tests_e2e/tests/lib/virtual_machine_extension_client.py, extension names in tests_e2e/tests/lib/identifiers.py, SSH operations on tests_e2e/tests/lib/ssh_client.py, etc

Ping me when you get to specific items and we can figure out where they should go.